File: spring/sentimental/mario/more/mario.py
#!/usr/bin/env python3

# Print out a double half-pyramid of a specified height

import cs50

# Get height from the user
while True:
    height = cs50.get_int("Height: ")
    if height >= 1 and height <= 8:
        break

# Print double half-pyramid
# For each layer
for i in range(1, height + 1):

    # Print left half-pyramide
    # Print spaces
    print(" " * (height - i), end="")
    # Print hashes
    print("#" * i, end="")

    # Print gap
    print(" " * 2, end="")

    # Print right half-pyramide
    # Print hashes
    print("#" * i, end="")

    # Print new line
    print()

File: spring/mario/more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Get height between 1 and 8 from the user
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while ((height < 1) || (height > 8));

    // Repeat from row #1 up to row #height
    for (int row = 1; row <= height; row++)
    {
        // Print #(height - rows) spaces 
        for (int j = 1; j <= height - row; j++)
        {
            printf(" ");
        }
    
        // Print #row hashes
        for (int j = 1; j <= row; j++)
        {
            printf("#");
        }

        // Print #2 spaces
        printf("  ");

        // Print #row hashes
        for (int j = 1; j <= row; j++)
        {
            printf("#");
        }

        // Print new line
        printf("\n");
    }
}
